### ISSUE TITLE
Brainomics is down

### ISSUE THREAD

**Issue by kchawla-pi** [nilearn/nilearn#1935]
@DimitriPapadopoulos   Brainomics seems to be down since Monday. WHen will it be back up?

**Comment by DimitriPapadopoulos** [comment]
I've let the IT team know. Most probably too many requests again, hence too many ZIP files generated, leading to a disk full situation. Strange, I was under the impression that disk usage was being monitored by the IT team - perhaps there were too many requests in a very short a time so they cannot react.

I really need to move Brainomics/Localizer data elsewhere as discussed in #1925.

**Comment by DimitriPapadopoulos** [comment]
Ah, the IT team have just written me the server should be back online.

**Comment by DimitriPapadopoulos** [comment]
The `brainomics.cea.fr` will be down from 2019-03-29 afternoon (CET) to 2019-04-01 evening or even 2019-04-02 morning (CEST).

**Comment by bthirion** [comment]
@kchawla-pi could we work on Monday on the replacement of the brainomics site by something more reliable ?

**Comment by bthirion** [comment]
@DimitriPapadopoulos  Thx for letting us know bzw.

**Comment by kchawla-pi** [comment]
@bthirion Yes.

**Comment by DimitriPapadopoulos** [comment]
By the way I keep getting warnings from the server, at least daily:
```/: 81% used (51GB/63GB) (80%) : WARNING```
As you can see this an old server with a very small disk, which explains the recurrent failures to generate a ZIP file.

* Since _nilearn_ needs maps only, you may extract them and add them to NeuroVault - if that's not already done. Just a request, please cite this paper:
https://github.com/nilearn/nilearn/issues/1925#issue-411592219
* Then I need to convert the whole dataset to NIfTI and publish it somewhere else, [perhaps _osf.io_ after all (but on the European server)](https://github.com/nilearn/nilearn/issues/1925#issuecomment-465495346).

**Comment by kchawla-pi** [comment]
@DimitriPapadopoulos Sorry didn't notice your last comment. Thanks for that. Any progress on the osf.io thing?

**Comment by DimitriPapadopoulos** [comment]
I need to convert the dataset to BIDS first, I probably won't have time before the summer. However since Nilearn needs only statistical maps, you can fetch them from [Neurovault](https://neurovault.org/collections/4341/) of the `brainomics.cea.fr` server.

**Comment by kchawla-pi** [comment]
>  As you can see this an old server with a very small disk,

Disks are inexpensive these days, have the admins considered upgrading?

ALso, Brainomics maintenance was supposed to be completed by Tueday latest. It hasn't come back up. When will it?

**Comment by DimitriPapadopoulos** [comment]
Disk are inexpensive indeed. Upgrading remains expensive.

**Comment by DimitriPapadopoulos** [comment]
I think I have found some help to convert Brainomics/Localizer to BIDS and push to OSF and/or Amazon. I'll keep you posted.

**Comment by bthirion** [comment]
Great, thx.

**Comment by DimitriPapadopoulos** [comment]
It appears we already have written some code to use alternative sources for Brainomics/Localizer:
https://github.com/AGrigis/nilearn/commits/master
We will try to have the code evolve too.

**Comment by kchawla-pi** [comment]
Hi @DimitriPapadopoulos 
Would the Brainomics server be back up on Monday morning? We have a coding sprint starting on Monday.
Also, is there a formal place to notify Brainomics staff? Or is it okay with you if I ping you here?

**Comment by DimitriPapadopoulos** [comment]
We are finalizing the conversion to BIDS format and will soon move the Brainomics/Localizer to _osf.io_. I do have an issue with _osf.io_ though, perhaps someone can help me:
We have thousands of files, I cannot upload them manually. I cannot find isntructions to automate uploading using a script on the OSF site. Is that not possible? If not _osf.io_ is probably not a good idea for storing data, except maybe for projects that use/produce almost no data. Of course I could put the dataset on Amazon S3, but that's way too expensive.

**Comment by emdupre** [comment]
`osfclient` uses an older version of the OSF API. You can see the current documentation here: https://developer.osf.io/ 

I'm also sharing a [snippet we used to download OSF keys](https://gist.github.com/emdupre/3cb4d564511d495ea6bf89c6a577da74) -- it might give a general idea for how to interact with the API !

**Comment by DimitriPapadopoulos** [comment]
@emdupre Thank you for sharing the snippet. Yet _osfclient_ might be the easiest way for me to upload the Brainomics/Localizer dataset in the short term: no need to write a Python script, just use the existing script. Is the older version of the OSF API still active, or has it been completely retired?

**Comment by emdupre** [comment]
I found it failed for me, since the way data is internally stored on OSF changed between the v1 and v2 release. I haven't used the API to upload data, though, only download. So I'm not sure how that functionality would be affected.

**Comment by DimitriPapadopoulos** [comment]
That said the v2 API seems rather easy. Perhaps you have an answer to some of the following questions?

Do I call _requests.put()_ instead of _requests.get()_ to upload data?

The _requests.put()_ isn't precisely well-documented:
``` python
>>> r = requests.put('https://httpbin.org/put', data = {'key':'value'})
```
Any hint how to prepare the _data_ argument of _requests.put()_? Do I need to send optional JSON data using the _json_ argument?

**Comment by DimitriPapadopoulos** [comment]
By the way, the OSF APIv2 Documentation still refers to [_osfclient_](https://developer.osf.io/#tag/External-Libraries). Perhaps has it been [ported to v2](https://github.com/osfclient/osfclient/search?q=v2) lately...

**Comment by kchawla-pi** [comment]
@DimitriPapadopoulos Hey, the Localizer datasets are unavailable for the past 3 days. I assume this is because you guys are moving the data to OSF. WHat's the timeline on that so we can change the download URL's in Nilearn's fetchers?

**Comment by DimitriPapadopoulos** [comment]
Concerning _brainomics.cea.fr_, I guess it's again a disk full condition as usual. Too many requests creating, too many ZIP files filling the disk.

Note that May is a month with lots of official holidays here, plus we have to use our days of vacation before the end of May. Lots of things just cannot happen before the end of May. We will probably be moving the data to OSF in June.

**Comment by DimitriPapadopoulos** [comment]
It's probably more than a disk full condition. Some process must have crashed. I'll notify the IT team...

**Comment by kchawla-pi** [comment]
@DimitriPapadopoulos The localizer datasets are unavailable since past 2 days. :-) 
What's the update to moving the data to OSF?

**Comment by bthirion** [comment]
Dimitri is currently uploading the data to osf. That could be done soon.

**Comment by kchawla-pi** [comment]
Would that explain the URL redirect errors I am seeing?

**Comment by DimitriPapadopoulos** [comment]
Probably a server overload?

I think the conversion to BIDS is almost done. I need to double-check the data and then upload to OSF or perhaps [B2ACCESS](https://b2access.eudat.eu).

**Comment by kchawla-pi** [comment]
Thanks @DimitriPapadopoulos !
Also, what's your office number?

**Comment by DimitriPapadopoulos** [comment]
The dataset has been converted to BIDS but unfortunately it's very complicated to upload to the OSF server. It keeps timing out and maybe there is a limit on the amount of data per project. I'll have to contact OSF support and maybe choose a more reliable service such as [EUDAT B2ACCESS](https://b2access.eudat.eu/).

**Comment by emdupre** [comment]
There's a 5GB limit on the [size of individual files, but not on the project](https://help.osf.io/hc/en-us/articles/360019737894-FAQs#what-is-the-individual-file-size-limit), at least with default OSF storage. I'm not sure why you'd be seeing time-outs...

**Comment by kchawla-pi** [comment]
@emdupre 
Nail-biting, edgy ongoings here :-) : https://github.com/nilearn/nilearn/issues/1925 .

**Comment by DimitriPapadopoulos** [comment]
While data has been uploaded to OSF, we see 404 HTTP errors when running tests. It looks like he OSF server fails to serve some files.

**Comment by jeromedockes** [comment]
closed by #2097
